lib/vrl/compiler/src/expression/assignment.rs

```diff
@@ -145,6 +145,23 @@ impl Assignment {
 
         Self { variant }
     }
+
+    /// Get a list of targets for this assignment.
+    ///
+    /// For regular assignments, this contains a single target, for infallible
+    /// assignments, it'll contain both the `ok` and `err` target.
+    pub(crate) fn targets(&self) -> Vec<Target> {
+        let mut targets = vec![];
```

Comment:
You're allocating a zero-sized vector here and then pushing one, maybe two values into it. In the two case you can avoid going to the allocator twice extra with

```suggestion
        let mut targets = Vec::with_capacity(2);
```

Reply:
Good catch. It's worth pointing out (though not an excuse not to apply these optimizations) that all code in the `compiler` crate has (as the name implies) no impact on Vector's runtime performance, all VRL programs are compiled at boot-time.

Still, as I said, it's generally good practice to keep the compiler as performant as possible, even if it has negligible effect on Vector itself.

Resolved in [f24dfb7](https://github.com/vectordotdev/vector/pull/12546/commits/f24dfb7427b02d367905ab00817a783de1c2a2bb).